### PR TITLE
Flow 9641 adapt to new api specs

### PR DESCRIPTION
**Pull request by ArnaudGathy**

### CODE REVIEW

**Comment by ArnaudGathy** [comment]
As there are a lot of prettier changes, I will mark functions that changed

The main thing that changed is that the API doesn't respond with the cells name, it gives an [`Array<Array<string>>`](https://github.com/taktik/simple-spreadsheet-reader/pull/2/files#diff-a5906d9c3b7c1fcee6a2fdeec6e54abd06d6b18395e30fa8cd133bec178ef4c0R58) `containing the cells values, so I had to parse those value and add the cells name to keep compatbility with the old code

Also fixed the issue that we don't display cells after column Z, now it's compatible with multi-letter column indexes (AA, AB, etc)

File: src/SpreadsheetReader.ts [hunk 1]
```diff
@@ -101,64 +160,110 @@ export class SpreadsheetReader {
...
-			.map(elem => {
-				const parcedCell = /([A-Z]+)([0-9]+)/.exec(elem.cell)
-				if(parcedCell === null) throw Error('Error in spredsheet format')
-				const [cellId, coll, rows] = parcedCell
-				return Object.assign({ rows, coll, cellId, collNb: coll.charCodeAt(0) }, elem)
-			})
-		const maxRow = Number(search(cellsList, 'max_by([*], &rows).rows'))
-		const maxColl = String.fromCharCode(search(cellsList, 'max_by([*], &collNb).collNb'))
-		return {cellsList, rawJson, maxRow, maxColl}
+	private processSpreadsheet(parsedCells: IParsedCells): ISheetsData {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:183]: 👆

File: src/SpreadsheetReader.ts [hunk 2]
```diff
@@ -101,64 +160,110 @@ export class SpreadsheetReader {
...
+			// parseInt(coll, 36) parses the letters as a number, which can then be compared to each other to define which is the "highest" column letter
+			if (parseInt(coll, 36) > parseInt(highestColl, 36)) {
+				return coll
+			}
+			return highestColl
+		}, 'A')
+		return { cellsList, parsedCells, maxRow, maxColl }
+	}
+
+	private static getColumnLettersFromIndex(index: number): string {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:207]: 👆

File: src/SpreadsheetReader.ts [hunk 3]
```diff
@@ -101,64 +160,110 @@ export class SpreadsheetReader {
...
+		return String.fromCharCode(index + 65)
+	}
+
+	/*
+	 * Function to parse the array we receive from google API into an array of objects containing the cell name alongside the cell value
+	 * [["cellValue1", "cellValue2"]] => [{cell: "A1", value: "cellValue1"}, {cell: "B2", value: "cellValue2"}]
+	 * This function is made to process results from a request using majorDimension=ROWS (default value for majorDimension on the get/value request)
+	 * If the request is made with another value for majorDimension, this function will break
+	 */
+	private parseSheetValues(sheetValues: ISheetValue): ISheetsData {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:225]: 👆

File: src/SpreadsheetReader.ts [hunk 4]
```diff
@@ -101,64 +160,110 @@ export class SpreadsheetReader {
...
 	/**
-	 * Load spreadsheet data
+	 * Load spreadsheet cells values
 	 */
 	async loadSpreadsheetData(): Promise<void> {
 		try {
-			if( ! this.spreadsheetsId ) {
-				throw Error('Invalid spreadsheetsId')
-			}
+			const sheetPropertiesUrl = `https://sheets.googleapis.com/v4/spreadsheets/${this.spreadsheetsId}?key=${this.apiKey}`
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:242]: 👆

File: src/SpreadsheetReader.ts [hunk 5]
```diff
@@ -174,19 +279,10 @@ export class SpreadsheetReader {
...
-
-	/**
-	 * gel all lines of the spreadsheet in an array of array
-	 */
-	getAllLines(): Array<Array<string | undefined>>{
-		const cellsByRaws = groupBy(this.cellsList, (cell) => cell.rows)
-		const results: Array<Array<string | undefined>> = []
-		for(const i of SpreadsheetReader.numberGenerator(this.maxRow)){
-			results.push( SpreadsheetReader.formatColl(cellsByRaws[i] || [], this.maxColl))
+		const matchingCell = this.parsedCells.find(({ cell }) => cell === cellId)
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:282]: 👆

File: src/SpreadsheetReader.ts [hunk 6]
```diff
@@ -174,19 +279,10 @@ export class SpreadsheetReader {
 	 */
 	getCellValue(cellId: string, page = 0): string | undefined {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:280]: 👆

File: src/SpreadsheetReader.ts [hunk 7]
```diff
@@ -101,64 +160,110 @@ export class SpreadsheetReader {
...
+			)
+		)
+		return this.processSpreadsheet(parsedValues)
 	}
 
 	/**
-	 * Load spreadsheet data
+	 * Load spreadsheet cells values
 	 */
 	async loadSpreadsheetData(): Promise<void> {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:240]: 👆

File: src/SpreadsheetReader.ts [hunk 8]
```diff
@@ -249,83 +345,76 @@ export class SpreadsheetReader {
...
 		return table
 	}
 
-	protected static formatColl(cells: Array<SpredsheedCell>, maxColl: string): Array<string | undefined>{
-		const cellsByColl = groupBy(cells, (cell) => cell.coll)
-		const results: Array<string> = []
-		for(const  i of SpreadsheetReader.lettersGenerator(maxColl)){
-			const cell: any = (cellsByColl[i] || [{ value: undefined}])[0]
-			results.push(cell.value)
+	private static *lettersGenerator(maxLetters: string): Generator<string> {
```
Comment (ArnaudGathy) [inline src/SpreadsheetReader.ts:366]: 👆

File: package.json [hunk 9]
```diff
@@ -23,7 +23,6 @@
   "author": "Taktik",
   "license": "ISC",
   "dependencies": {
-    "jmespath": "^0.15.0",
```
Comment (ArnaudGathy) [inline package.json:26]: No longer used with the changes I made, as the response is now pretty easy to parse we don't need a complicated query lib for JSON

**Review by Azevedo-Dani — APPROVED** [review]

**Review by lmancel — APPROVED** [review]

**Pull request merged**